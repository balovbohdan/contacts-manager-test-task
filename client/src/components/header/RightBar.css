.main {
    display: flex;
    align-items: center;
    justify-content: flex-end;

    width: 50%;
}

.btn {
    cursor: pointer;
    padding: 5px 10px;
    transition: var(--trans);
    border-radius: var(--border-radius);

    color: var(--color-white);
}

.btn__reg {}
.btn__login {}
.btn__about {}

.btn:hover {
    background: rgba(255, 255, 255, .1);
}