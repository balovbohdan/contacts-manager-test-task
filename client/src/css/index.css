@import url(reset.css);
@import url(fonts.css);
@import url(titles.css);
@import url(constants.css);
@import url(main.css);