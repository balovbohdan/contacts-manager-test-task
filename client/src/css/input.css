.label {
    font-size: .9rem;
}

.input {
    width: 100%;
    padding: 5px 0;
    border-bottom: 2px solid var(--color-yellow);

    font-size: 1.5rem;
    color: var(--color-white);
}

@media (max-width: 600px) {
    .input {
        font-size: 1.2rem;
    }

    .label {
        font-size: .8rem;
    }
}

@media (max-width: 400px) {
    .input {
        font-size: 1.1rem;
    }

    .label {
        font-size: .7rem;
    }
}