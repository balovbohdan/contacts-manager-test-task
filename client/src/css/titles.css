h1, h2, h3, h4, h5, h6 {
    font-size: 100%;
    font-family: var(--font-main);
}

h1 {
    font-size: 40px;
}

h2 {
    font-size: 35px;
}

h3 {
    font-size: 30px;
}

h4 {
    font-size: 25px;
}

h5 {
    font-size: 20px;
}

h6 {
    font-size: 17px;
}

@media (max-width: 550px) {
    h1 {
        font-size: 35px;
    }

    h3 {
        font-size: 25px;
    }
}

@media (max-width: 400px) {
    h1 {
        font-size: 30px;
    }

    h3 {
        font-size: 20px;
    }
}