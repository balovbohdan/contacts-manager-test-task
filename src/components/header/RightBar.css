.main {
    display: flex;
    align-items: center;
    justify-content: flex-end;

    width: 50%;
}

.btn {
    cursor: pointer;
    padding: 5px 10px;
    border-radius: 3px;
    transition: var(--trans);

    color: var(--color-white);
}

.btn:hover {
    background: rgba(255, 255, 255, .1);
}