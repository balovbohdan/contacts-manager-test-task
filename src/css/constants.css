:root {
    --color-red: #FB4B4E;
    --color-blue: #16324F;
    --color-white: #F3F7F0;
    --color-yellow: #FDE74C;
    --color-light-blue: #236098;

    --font-main: Roboto, serif;
    --font-marker: 'Permanent Marker', serif;

    --border-radius: 3px;
    --header-height: 60px;
    --trans: all .12s ease-in;
}