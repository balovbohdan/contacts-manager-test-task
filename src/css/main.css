body {
    background: var(--color-light-blue);
}