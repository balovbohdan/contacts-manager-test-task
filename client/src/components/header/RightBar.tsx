import * as React from 'react';
import classnames from 'classnames';

import * as css from './RightBar.css';

type Props = {
    toggleRegWindow:()=>void;
    toggleLoginWindow:()=>void;
};

export const RightBar = (props:Props) =>
    <div className={css.main}>
        <AboutBtn/>
        <RegBtn onClick={props.toggleRegWindow}/>
        <LoginBtn onClick={props.toggleLoginWindow}/>
    </div>;

const Btn = ({onClick, children, className}) => {
    const classNames = classnames(css.btn, className);

    return (
        <div className={classNames} onClick={onClick}>
            {children}
        </div>
    );
};

const LoginBtn = ({onClick}) =>
    <Btn onClick={onClick} className={css.btn__login}>
        Login
    </Btn>;

const RegBtn = ({onClick}) =>
    <Btn onClick={onClick} className={css.btn__reg}>
        Register
    </Btn>;

const AboutBtn = () =>
    <Btn onClick={()=>{}} className={css.btn__about}>
        About
    </Btn>;