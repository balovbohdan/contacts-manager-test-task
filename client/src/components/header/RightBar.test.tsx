import * as React from 'react';
import {mount, shallow} from 'enzyme';

import {RightBar} from './RightBar';
import * as css from './RightBar.css';

describe('RightBar of Header', () => {
    it('Should match snapshot', () => {
        const toggleRegWindow = jest.fn();
        const toggleLoginWindow = jest.fn();

        const bar = shallow(
            <RightBar
                toggleRegWindow={toggleRegWindow}
                toggleLoginWindow={toggleLoginWindow}/>
        );

        expect(bar).toMatchSnapshot();
    });

    it('Should react on btns clicks', () => {
        const toggleRegWindow = jest.fn();
        const toggleLoginWindow = jest.fn();

        const bar = mount(
            <RightBar
                toggleRegWindow={toggleRegWindow}
                toggleLoginWindow={toggleLoginWindow}/>
        );

        bar.find(`.${css.btn}.${css.btn__reg}`).simulate('click');
        bar.find(`.${css.btn}.${css.btn__login}`).simulate('click');

        expect(toggleRegWindow).toBeCalledTimes(1);
        expect(toggleLoginWindow).toBeCalledTimes(1);
    });
});