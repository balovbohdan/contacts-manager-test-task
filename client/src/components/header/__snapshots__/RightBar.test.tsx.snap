// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightBar of Header Should match snapshot 1`] = `
<div
  className="main"
>
  <AboutBtn />
  <RegBtn
    onClick={[MockFunction]}
  />
  <LoginBtn
    onClick={[MockFunction]}
  />
</div>
`;
